types/stream-to-string/index.d.ts

```diff
@@ -0,0 +1,20 @@
+// Type definitions for stream-to-string 1.1
+// Project: https://github.com/jasonpincin/stream-to-string
+// Definitions by: BendingBender <https://github.com/BendingBender>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+export = streamToString;
+
+// tslint:disable:unified-signatures
+
+declare function streamToString(stream: NodeJS.ReadableStream): Promise<string>;
+declare function streamToString(stream: NodeJS.ReadableStream, cb: Callback): Promise<string>;
```

Comment:
If you just make this `cb?: Callback` then you can get rid of the first overload and (I think) also the tslint disable.

Reply:
The problem is that I don't want to mix the signature of the `enc` parameter with the `cb` parameter. I find this version much less comprehensible than the current one:
```
declare function streamToString(
    stream: NodeJS.ReadableStream,
    enc?: Callback | string,
    cb?: Callback
): Promise<string>;
```
It is just not clear what happens here and what parameters go where. I know that tecnically, this is the same.

Reply:
I wouldn't suggest combining all three into one, but just the first two:
```ts
declare function streamToString(stream: NodeJS.ReadableStream, cb?: Callback): Promise<string>;
declare function streamToString(
    stream: NodeJS.ReadableStream,
    enc: string,
    cb?: Callback
): Promise<string>;
```

This way it's clear that there's a version with one param, one with two where the second is cb, one with two where the second is enc, and one with three.

Reply:
Oh, I didn't try this one. Fixed.